File: Backend/Models/UserManagement.cs
namespace ActoEngine.WebApi.Models;

public class UserDto
{
    public int UserId { get; set; }
    public string Username { get; set; } = default!;
    public string? FullName { get; set; }
    public bool IsActive { get; set; }
    public string? RoleName { get; set; }
    public int? RoleId { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
}

public class CreateUserRequest
{
    public string Username { get; set; } = default!;
    public string Password { get; set; } = default!;
    public string? FullName { get; set; }
    public int RoleId { get; set; }
}

public class UpdateUserRequest
{
    public int UserId { get; set; }
    public string? FullName { get; set; }
    public int RoleId { get; set; }
    public bool IsActive { get; set; }
}

public class ChangePasswordRequest
{
    public int UserId { get; set; }
    public string NewPassword { get; set; } = default!;
}

public class UserWithPermissionsDto
{
    public UserDto User { get; set; } = default!;
    public List<string> Permissions { get; set; } = [];
}

File: Frontend/src/types/user-management.ts
// User types
export interface UserDto {
  userId: number;
  username: string;
  fullName?: string;
  isActive: boolean;
  roleName?: string;
  roleId?: number;
  createdAt: string;
  updatedAt?: string;
}

export interface CreateUserRequest {
  username: string;
  password: string;
  fullName?: string;
  roleId: number;
}

export interface UpdateUserRequest {
  userId: number;
  fullName?: string;
  roleId: number;
  isActive: boolean;
}

// Role types
export interface Role {
  roleId: number;
  roleName: string;
  description?: string;
  isSystem: boolean;
  isActive: boolean;
  createdAt: string;
  updatedAt?: string;
}

export interface Permission {
  permissionId: number;
  permissionKey: string;
  resource: string;
  action: string;
  description?: string;
  category?: string;
  isActive: boolean;
  createdAt: string;
}

export interface PermissionGroupDto {
  category: string;
  permissions: Permission[];
}

export interface RoleWithPermissions {
  role: Role;
  permissions: Permission[];
}

export interface CreateRoleRequest {
  roleName: string;
  description?: string;
  permissionIds: number[];
}

export interface UpdateRoleRequest {
  roleName: string;
  description?: string;
  isActive: boolean;
  permissionIds: number[];
}

export interface UserDetailResponse {
  user: UserDto;
  permissions: string[];
}
